documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -92,73 +144,468 @@ for (Object book : objectList) {
+In any case, having a single write lock cluster-wise can lead to some degree of contention under heavy writing.
+
+Infinispan support shared indexes leveraging the <<integrations.adoc#integrations:directory-provider,Infinispan Directory Provider>>, which stores indexes in a separate set of caches. Two index managers are available
+to use shared indexes: InfinispanIndexManager and AffinityIndexManager.
+
+====== Effect of the index mode
+
+Shared indexes should not use the `ALL` index mode since it'd lead to redundant indexing: since there is a single index cluster wide, the entry would get indexed when inserted via Cache API, and another time
+when Infinispan replicates it to another node.
+
+====== InfinispanIndexManager
+
+This index manager uses the <<integrations.adoc#integrations:directory-provider,Infinispan Directory Provider>>, and is suitable for creating shared indexes. Index mode should be set to `LOCAL`.
+
+Configuration:
+
+[source,xml]
+----
+<distributed-cache name="default" >
+    <indexing index="LOCAL">
+        <property name="default.indexmanager">
+            org.infinispan.query.indexmanager.InfinispanIndexManager
+        </property>
+        <!-- optional: tailor each index cache -->
+        <property name="default.locking_cachename">LuceneIndexesLocking_custom</property>
+        <property name="default.data_cachename">LuceneIndexesData_custom</property>
+        <property name="default.metadata_cachename">LuceneIndexesMetadata_custom</property>
+    </indexing>
+</distributed-cache>
+
+<!-- Optional -->
+<replicated-cache name="LuceneIndexesLocking_custom">
+    <indexing index="NONE" />
+</replicated-cache>
+
+<!-- Optional -->
+<replicated-cache name="LuceneIndexesMetadata_custom">
+    <indexing index="NONE" />
+</replicated-cache>
+
+<!-- Optional -->
+<distributed-cache name="LuceneIndexesData_custom">
+    <indexing index="NONE" />
+</distributed-cache>
+
+----
+
+Indexes are stored in a set of clustered caches, called by default _LuceneIndexesData_, _LuceneIndexesMetadata_ and _LuceneIndexesLocking_.
+
+The _LuceneIndexesLocking_ cache is used to store Lucene locks, and it is a very small cache: it will contains one entry per entity (index).
+
+The _LuceneIndexesMetadata_ cache is used to store info about the logical files that are part of the index, such as names, chunks and sizes and it is also
+small in size.
+
+The _LuceneIndexesData_ cache is the where most of the index is located: it is much bigger then the other two but should be smaller than the data in the cache itself, thanks to Lucene
+efficient storing techniques.
+
+The configuration above is changing the name of those 3 caches and redefining their configuration. Reasons to do so is performance tuning: since two of the 3 caches are small
+and heavily accessed, it makes sense to make them replicated, while the largest cache is configured as distributed. Manually redefining the index caches also
```

Comment:
Since you mentioned, I think it became the default very recently, let me double check